src/EditorFeatures/CSharpTest/AutomaticCompletion/AutomaticBraceCompletionTests.cs

```diff
@@ -1348,7 +1342,506 @@ public void CurlyBraceFormatting_InsertsCorrectNewLine()
             CheckReturn(session.Session, 4, result: "class C\r{\r\r}");
         }
 
-        internal static Holder CreateSession(string code, Dictionary<OptionKey2, object> optionSet = null)
+        [WorkItem(50275, "https://github.com/dotnet/roslyn/issues/50275")]
+        [WpfFact, Trait(Traits.Feature, Traits.Features.AutomaticCompletion)]
+        public void WithInitializer_OpenBraceOnSameLine_Enter()
```

Comment:
💡 This test can cover both cases. Start by defining it as a theory:

```suggestion
        [WorkItem(50275, "https://github.com/dotnet/roslyn/issues/50275")]
        [WpfTheory, CombinatorialData, Trait(Traits.Feature, Traits.Features.AutomaticCompletion)]
        public void WithInitializer_Enter(bool bracesOnNewLine)
```

Then create locals for both `expectedForSameLine` and `expectedForNewLine` and set the option and assertion according to the argument and correct local, respectively. This test form is not only condensed, but it more clearly shows the impact of the distinct option values considering the starting `code` is known to be identical for both tests.

This suggestion applies for each pair added in this PR.

Reply:
@sharwell Thanks! Fixed.